ngraph/core/src/op/util/gather_base.cpp

```diff
@@ -52,26 +52,36 @@ void op::util::GatherBase::validate_and_infer_types()
             axis_pshape);
     }
 
-    int64_t batch_dims = get_batch_dims(); // will not be converted to positive if axis is not set
-    if (is_axis_set())
+    int64_t batch_dims = m_batch_dims;
+    if (batch_dims < 0 && indices_rank.is_static())
+    {
+        batch_dims += indices_rank.get_length();
+    }
+
+    bool axis_is_set = false;
+    if (get_constant_from_source(input_value(2)))
+        axis_is_set = true;
+
+    if (axis_is_set)
     {
         int64_t axis = get_axis();
+
+        // batch_dims, axis both can be positive by default or after normalization if data_rank &
+        // indices_rank are static. If at least one of them is negative we cannot check their
+        // consistency.
         NODE_VALIDATION_CHECK(this,
-                              batch_dims <= axis,
+                              batch_dims <= axis || batch_dims < 0 || axis < 0,
                               "The batch_dims <= axis. But instead got: batch_dims = ",
                               batch_dims,
                               ", axis = ",
                               axis);
 
-        if (data_rank.is_static())
-        {
-            NODE_VALIDATION_CHECK(this,
-                                  axis >= 0 && axis < data_rank.get_length(),
-                                  "The axis must be >= 0 and < data_rank. But instead got axis = ",
-                                  axis,
-                                  " data_rank = ",
-                                  data_rank.get_length());
-        }
+        NODE_VALIDATION_CHECK(this,
+                              axis < data_rank.get_length() || data_rank.is_dynamic(),
+                              "The axis must be >= 0 and < data_rank. But instead got axis = ",
```

Comment:
Please, update the message to correspond the check.

Reply:
thanks! fixed